### DISCUSSION TITLE
Allow issues to be assigned to author of pull-requests that mentions the issue

### DISCUSSION THREAD

**Discussion by augustoproiete** [community/community#5016] [Projects and Issues]
Scenario:

- **Alice** and **Bob** are contributors to open-source projects (external to the repo/organization)
- **Alice** creates issue `123`
- **Bob** later creates pull request `456` that resolves issue `123`

Given that Bob is effectively resolving issue `123` it would make sense for users that can manage issues to be able to assign issue `123` to **Bob** in order to signal to other contributors that work on the issue is already in-progress / issue already has an owner.

Unfortunately this is not possible today. The only way to assign the issue to **Bob** is to ask **Bob** to add a comment to the issue, so that his user becomes available as a possible assignee.

**Comment by michael-hawker** [comment]
(Is there no reply feature, I can only add an 'answer'?)

Yeah, we have this same problem. In general, it'd be nice to be able to assign an issue to anyone. Here's a slightly different, but similar scenario:

- **Bob** adds a feature to the repo.
- **Alice** finds an issue and reports it in `124`

I know **Bob** worked on that feature; so I want to assign the issue to **Bob** to look at, but can't if he's not a member of the repo/org.

Duplicate/Related of https://github.com/github/feedback/discussions/2897

**Comment by inomdzhon** [comment]
Absolutely same problem!

![image](https://github.com/community/community/assets/5850354/51abd374-6ca4-4bb4-b0e7-3aa6022ae6e0)
